### PR TITLE
Feature/momza 784 remove sms references in message sets

### PR DESCRIPTION
**Pull request by dubesoftware**
My Python plugin for VS Code is formatting for PEP-8 as well.

### CODE REVIEW

File: seed_services_cli/main.py [hunk 1]
```diff
@@ -35,7 +36,7 @@ class ci(object):
 @click.group(name="seed-services-cli")
 @click.version_option()
 @click_config.wrap(module=config, sections=('hub', 'identity_store',
-                   'stage_based_messaging', 'auth', 'ci'))
+                                            'stage_based_messaging', 'auth', 'cli'))
```
Comment (rudigiesler) [inline seed_services_cli/main.py]: I think the `ci` to `cli` change is incorrect. I think the `ci` section is for the control interface.

File: seed_services_cli/stage_based_messaging.py [hunk 2]
```diff
@@ -167,6 +167,37 @@ def messages_import(ctx, csv, json):
...
+        raise click.UsageError("please specify either --csv or --json")
+    api = get_api_client(ctx.obj.stage_based_messaging.api_url,
+                         ctx.obj.stage_based_messaging.token)
+    if csv:
+        for msg in messages_from_csv(csv):
+            if msg["binary_content"] is not None and \
+                    msg["binary_content"] != "":
+                msg = create_binarycontent(api, msg)
+            click.echo("Updating message to messageset %(messageset)s." % msg)
+            api.update_message(msg)
```
Comment (rudigiesler) [inline seed_services_cli/stage_based_messaging.py]: I think the `update_message` function in the stage based messaging client takes two parameters, the ID of the message, and the data to use to update the message.

File: seed_services_cli/stage_based_messaging.py [hunk 3]
```diff
@@ -167,6 +167,37 @@ def messages_import(ctx, csv, json):
...
+    if json:
+        for msg in messages_from_json(json):
+            if msg["binary_content"] is not None and \
+                    msg["binary_content"] != "":
+                msg = create_binarycontent(api, msg)
+            click.echo("Updating message to messageset %(messageset)s." % msg)
+            api.update_message(msg)
```
Comment (rudigiesler) [inline seed_services_cli/stage_based_messaging.py]: This seems to be some repeated code. If the data returned from `messages_from_csv` and `message_from_json` is in the same format, maybe we can do something like:
```
if csv:
    messages = messages_from_csv(csv)
elif json:
    messages = messages_from_json(json)

for message in messages:
    ....
```

Comment (dubesoftware) [reply seed_services_cli/stage_based_messaging.py]: I agree. I will call the messages msgs though, as there already is a messages(...) function defined.

Comment (rudigiesler) [reply seed_services_cli/stage_based_messaging.py]: 👍 Variable names can be whatever works best, it was just a guideline to show the general code structure that I was suggesting :)

File: seed_services_cli/stage_based_messaging.py [hunk 4]
```diff
@@ -167,21 +167,66 @@ def messages_import(ctx, csv, json):
...
+    if csv:
+        messages = messages_from_csv(csv)
+    elif json:
+        messages = messages_from_json(json)
+    params = {}
+    for message in messages:
+        params["messageset"] = message["messageset"]
+        params["lang"] = message["lang"]
+        params["sequence_number"] = message["seqno"]
+        results = list(api.get_messages(params=params)['results'])
```
Comment (erikh360) [inline seed_services_cli/stage_based_messaging.py]: You will need to mock this API lookup in your tests.

File: seed_services_cli/stage_based_messaging.py [hunk 5]
```diff
@@ -167,21 +167,66 @@ def messages_import(ctx, csv, json):
...
+            raise click.UsageError("Message with messageset: %s, language: %s, \
+            sequence number: %s not found.") % \
+                (message["messageset"], message["lang"], message["seqno"])
+    for message in messages:
+        if message["binary_content"] is not None and \
+                message["binary_content"] != "":
+            message = create_binarycontent(api, message)
+        click.echo("Updating message to messageset %(messageset)s." %
+                   message)
+        api.update_message(message)
```
Comment (erikh360) [inline seed_services_cli/stage_based_messaging.py]: Over here you will need to send the message['id'] to the `update_message` function.

File: seed_services_cli/tests/test_stage_based_messaging.py [hunk 6]
```diff
@@ -175,3 +181,40 @@ def test_message_import_csv(self, create_patch):
...
+            in result.output)
+
+    def test_message_update_error_with_no_param(self):
+        result = self.runner.invoke(cli, ['sbm-messages-update'])
+        self.assertEqual(result.exit_code, 2)
+        self.assertTrue(
+            'Please specify either --csv or --json.' in result.output)
+
+    @patch('seed_services_client.StageBasedMessagingApiClient.update_message')
+    def test_message_update_csv(self, create_patch):
```
Comment (erikh360) [inline seed_services_cli/tests/test_stage_based_messaging.py]: Rename `create_patch` to `update_patch` and add the api lookup patch here.

File: seed_services_cli/tests/test_stage_based_messaging.py [hunk 7]
```diff
@@ -175,3 +181,40 @@ def test_message_import_csv(self, create_patch):
...
+            b'messageset,sequence_number,lang,text_content,binary_content\n')
+        csv_file.write(b'1,2,eng_ZA,"message text",""')
+        csv_file.flush()
+
+        result = self.runner.invoke(
+            cli, ['sbm-messages-update', '--csv={0}'.format(csv_file.name)])
+        csv_file.close()
+
+        self.assertEqual(result.exit_code, 0)
+        create_patch.assert_called_with({
```
Comment (erikh360) [inline seed_services_cli/tests/test_stage_based_messaging.py]: This function should be called with the `id` that you put in the API lookup mock too.

**Review by erikh360 — COMMENTED** [review]
Looking good, please have a look at my comments.

File: seed_services_cli/tests/test_stage_based_messaging.py [hunk 8]
```diff
@@ -175,3 +181,113 @@ def test_message_import_csv(self, create_patch):
...
+            'binary_content': ''
+        })
+
+    @patch('seed_services_client.StageBasedMessagingApiClient.update_message')
+    @patch('seed_services_client.StageBasedMessagingApiClient.get_messages')
+    def test_message_update_csv_no_messages(self, get_patch, update_patch):
+        """Test that result output is empty when no message is returned in a lookup.
+        """
+        get_patch.return_value = {'results': [
+            {
```
Comment (erikh360) [inline seed_services_cli/tests/test_stage_based_messaging.py]: This empty `dict` will means there is still one record in results, rather make it a empty list then you can assert if `result.exit_code` is 2 and that `Message not found` is in the output.

File: seed_services_cli/tests/test_stage_based_messaging.py [hunk 9]
```diff
@@ -175,3 +181,113 @@ def test_message_import_csv(self, create_patch):
...
+        csv_file.write(
+            b'messageset,sequence_number,lang,text_content,binary_content\n')
+        csv_file.write(b'1,2,eng_ZA,"message text",""')
+        csv_file.flush()
+
+        result = self.runner.invoke(
+            cli, ['sbm-messages-update', '--csv={0}'.format(csv_file.name)])
+        csv_file.close()
+
+        self.assertTrue('Error: Multiple messages found' in result.output)
```
Comment (erikh360) [inline seed_services_cli/tests/test_stage_based_messaging.py]: I would also assert the `exit_code` here.

**Review by erikh360 — COMMENTED** [review]
Please have a look at the two comments. Please also make sure flake8 passes.

Looking good otherwise.

**Review by KaitCrawford — COMMENTED** [review]
It's looking great. But I think a test that confirms it works with json input would be good as well

**Comment by dubesoftware** [comment]
@KaitCrawford Not sure about the JSON test after playing with it a bit. What would that entail in terms of the get_patch for instance? @erikh360 do you have some idea as to how to address/ implement Kaitlyn's suggestion?

**Comment by erikh360** [comment]
@dubesoftware you can make a copy of the `test_message_update_csv` test and change it to json. So instead of creating a csv file and sending it to the command you will create a json file to send.

**Review by KaitCrawford — APPROVED** [review]
:cake: great job! :+1:

**Comment by dubesoftware** [comment]
Yaaaas! Thanks @erik @KaitCrawford

**Pull request merged**